### DISCUSSION TITLE
The activity function 'Dns01Precondition' failed

### DISCUSSION THREAD

**Discussion by ebowdenqa** [shibayan/appservice-acmebot#284] [Q&A] [answered]
**Describe the bug**
The activity function 'Dns01Precondition' failed.  DNS zone(s) are not found. DnsNames = *.dev.myrootdomain.com.

I'm trying to provision a wildcard certificate, but it is also not working when running the Dns01Precondition on a standard domain, such as dev.myrootdomain.com.

**To Reproduce**
1. Within the Azure function, navigate to the function Dns01Precondition
2. Choose Code + Test
3. For the input, enter a domain valid for the web application, such as ['myrootdomain.com']
4. Press Run
5. 2021-02-12T14:36:45.378 [Error] Executed 'Dns01Precondition' (Failed, Id=58c8a8b0-81e6-42dd-aa2f-b6549c4082da, Duration=166ms)DNS zone(s) are not found. DnsNames = myrootdomain.com

The same error occurs if I use the UI found at https://func-myfunctionname-foag.azurewebsites.net/add-certificate

**Environment (please complete the following information):**
 - App Service OS: Windows
 - App Service Type: App Service Plan + Web App
 - Runtime Stack: .Net 4.8
 - Certificate Type: Wildcard.  Though, I have also tried standard certificates

**Additional context**
My app service plan and web app are in resource group A, subscription X.  My DNS zones are in resource group B, subscription Y.  Per the instructions, I assigned the service account for my Azure function as Web Plan Contributor and Web Site Contributor to resource group A.  I assigned the service account for my Azure function as DNS Zone Contributor to resource group B.

I waited over 30 minutes after assigning my service account as DNS Zone Contributor.

Any suggestions or advice for how to debug this?  I suspect my environment having more than one subscription is the difference between my setup and others.

**Comment by shibayan** [comment, marked as answer]
Make sure that Azure DNS is present in the same subscription as Acmebot.

Acmebot will set the same subscription to `Acmebot:SubscriptionId` by default.

**Reply by ebowdenqa** [reply]
Thank you @shibayan  .  Can the Azure function span multiple subscriptions?  For example, can I install the Acmebot in subscription A, where my Azure DNS is located, and it will work with my App Service and Web App, which are in subscription B?  ...assuming I have the permissions confgured, as described in the readme. 

Or does everything have to be in the same subscription?

**Reply by shibayan** [reply]
Acmebot can be deployed to a different subscription than the App Service / DNS Zone. In that case, you need to set `Acmebot:SubscriptionId` to the subscription ID where the App Service / DNS Zone resides.

The App Service and DNS Zone must always reside in the same subscription.

**Reply by ebowdenqa** [reply]
Thank you for clarifying.  Unfortunately, I cannot have my App Service and DNS Zone in the same subscription.

**Reply by shibayan** [reply]
In this case, please use the Key Vault version.
https://github.com/shibayan/keyvault-acmebot
